### DISCUSSION TITLE
cyclic refrenence

### DISCUSSION THREAD

**Discussion by liansghuaifan** [JanusGraph/janusgraph#2513] [Q&A]
i want to konw  Specific operation in  standardOporcessor or traversalOporcessor ，but analysis  not the same，so i wan  to get  
Specific operation from standardjanusgraphTx 
i need use standardjanusgraphTx  of janusfraph-core  in the standardOporcessor of gremlin-server, but it will cyclic dependedcy 
Does anyone offer ideas or help？

**Comment by li-boxuan** [comment]
What is your use case? Are you trying to develop gremlin-server source code?

**Reply by liansghuaifan** [reply]
是的，想在gremlin-server处拦截具体的操作，比如添加、删除、查询等，发现是 standardOporcessor or traversalOporcessor 两套不同的处理接口，不方便做统一拦截，所以想在事务处做拦截吗，从事务提交处获取具体的操作请求。

**Reply by li-boxuan** [reply]
So you want to intercept the operations received on the gremlin-server side. I am not familiar with gremlin-server thus could offer much help here :(  Have you checked out https://docs.janusgraph.org/basics/transaction-log/#transaction-log-use-cases and see if that helps with your use case? It does not record queries but does record all mutations.

**Reply by li-boxuan** [reply]
Another place you could seek help is the gremlin user mailing list.

**Reply by liansghuaifan** [reply]
好的，感谢回复！
